### PR TITLE
Pull request template

### PR DESCRIPTION
**Pull request by aaime**
This PR adds a pull request template. Hopefully it will improve the quality of PRs and make our life a bit easier.

Structure and introductory text shamlessly copied from QGIS one :-p

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+- [ ] If this is your first contribution, confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) and sent a CLA as necessary (not required for small changes, e.g., fixing typos in documentation)
+- [ ] There is a ticket in Jira describing the issue/improvement/feature
+- [ ] PR for bug fixes and small new features are presented as a single commit
+- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
+- [ ] New unit tests have been added covering the changes
+- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
+- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
+- [ ] Commits chaniging the UI, existing user workflows, or add new functionality, need to include documentation updates
```
Comment (bradh) [inline .github/PULL_REQUEST_TEMPLATE.md]: typo: changing

**Comment by ianturton** [comment]
looks good to me

**Comment by aaime** [comment]
@ianturton  "speeling"? :-D

**Comment by aaime** [comment]
Well let's merge, @bradh and others, further improvements on a later date are welcomed!

**Pull request merged**